### PR TITLE
gh-106368: Increase Argument Clinic test coverage

### PR DESCRIPTION
**Pull request by erlend-aasland**
Add tests for directives and destinations

* Issue: gh-106368

### CODE REVIEW

**Comment by erlend-aasland** [comment]
With this PR, we're up at 93% coverage (up 1% from 92%)

File: Lib/test/test_clinic.py [hunk 1]
```diff
@@ -416,6 +417,230 @@ def __init__(self):
...
+            printout test
+            [clinic start generated code]*/
+            test
+            /*[clinic end generated code: output=4e1243bd22c66e76 input=898f1a32965d44ca]*/
+        """)
+        generated = self.clinic.parse(block)
+        self.assertEqual(generated, expected)
+
+    def test_directive_preserve_twice(self):
+        err = "Can't have preserve twice in one block!"
```
Comment (AlexWaygood) [inline Lib/test/test_clinic.py:601]: I think this error message would read better if it were:

```suggestion
        err = "Can't have 'preserve' twice in one block!"
```

Maybe change this line, to make it so?

https://github.com/python/cpython/blob/9e6590b0978876de14587f528a09632b8879c369/Tools/clinic/clinic.py#L4583

**Review by AlexWaygood — COMMENTED** [review]
Looks good, just one nit:

File: Lib/test/test_clinic.py (hunk 1 above)
Comment (erlend-aasland) [reply Lib/test/test_clinic.py:601]: +1

There's a lot of places where we would like to use the `repr` in the error message. I started preparing a PR for normalising all the error messages yesterday.

Comment (erlend-aasland) [reply Lib/test/test_clinic.py:601]: Perhaps we can save this for that grand PR.

Comment (AlexWaygood) [reply Lib/test/test_clinic.py:601]: Sure. I generally dislike adding new tests that assert undesirable behaviour, even if it's only temporary (as it's all too easy to get distracted by something else and forget to follow up on it, or to realise that fixing the undesirable behaviour is harder than it seems)...

But in this case, I trust you to follow up on this :) (The undesirable behaviour is also pretty minor!)

**Review by AlexWaygood — APPROVED** [review]

File: Lib/test/test_clinic.py (hunk 1 above)
Comment (erlend-aasland) [reply Lib/test/test_clinic.py:601]: Here you are ;) gh-107615

**Pull request merged**